### DISCUSSION TITLE
SRS

### DISCUSSION THREAD

**Discussion by MelonigeMelone** [MelonigeMelone/xCalendar#7] [Dokumente]
# Software-Anforderungsspezifikation für „xCalendar“

## Softwareanforderungen:

- Einfacher Terminkalender mit kollaborativer Terminplanung
- Kalender oder Termine synchronisieren und teilen
- Aktivitätsbezogene Chats
- Ort pro Termin
- Automatische Navigation durch einfachen Knopfdruck
- Wetter-Funktion
- Benachrichtigung des Wetter am geplanten Termin
- Weitere Features folgen…

### 1.1. Übersicht

Diese Anwendung soll das Planen von Terminen mit zusätzlichen Freunden, Bekannten oder Arbeitskollegen vereinfachen. Man soll schnell und einfach sich auf einen Termin einigen können und jeder kann seinen Beitrag zu diesen Terminen leisten, indem der Ort geändert, der Zeitpunkt/Zeitraum oder gar jemand an einem Termin nicht teilnehmen kann.

### 1.2. Geltungsbereich

Diese Dokument behandelt zunächst das gesamte System, dass das Backend darstellt. Die Backend-Daten sollen zunächst in der Webanwendung verarbeitet werden.

### 1.3. Definitionen, Akronyme und Abkürzungen

API – Application Programming Interface: Eine Programmierschnittstelle zur Anbindung an ein System
 

### 2.1. Übersicht

Die Benutzer müssen einen Account erstellen, den sie auch wieder löschen können, wenn er nicht benötigt wird. Dadurch hat der Benutzer Zugriff auf seinen eigenen Kalender und kann sich auch die Kalender anderer Nutzer aus seiner Freundesliste anzeigen lassen. Dadurch soll man Termine erstellen können, und diese mit den Teilnehmern abstimmen können. Außerdem sollen die Teilnehmer dann über diesen Termin diskutieren können, um Ort, Zeit oder andere Bedingungen zu klären. Es können Termine vom Ersteller bearbeitet werden, oder Teilnehmer können eine Terminänderung vorschlagen, die dann der Ersteller annehmen oder ablehnen kann.

<img width="329" alt="image" src="https://github.com/MelonigeMelone/xCalendar/assets/52330056/3c36a07b-6979-4fe4-8fda-04c7d885f863">

### 2.2.	Aktivitätsbezogene Chats
Pro Termin gibt es einen Chat, indem man mit anderen Terminteilnehmern über einen Termin diskutieren kann, und Änderungen vorschlagen kann, wie zum Beispiel ein alternativer Ort oder ein neues Datum. Dabei können die Teilnehmer Änderungen vorschlagen und schicken damit einen Vorschlag an den Ersteller. Dabei kann der Ersteller die Änderung annehmen oder auch ablehnen. Jedoch gibt es auch die Möglichkeit mit einer Abstimmung, Änderungen vornehmen zu können. 

![image](https://github.com/MelonigeMelone/xCalendar/assets/52330056/0bdddd90-fd57-4932-8871-7e146993f858)

### 3.1 Login sequenzdiagramm
![image](https://github.com/MelonigeMelone/xCalendar/assets/52330056/ef63123a-16cd-47f5-8ae7-372bf70a702b)

### 4.1 Registrieren eines neuen Benutzers
![Aktivitätsdiagramm Login](https://github.com/MelonigeMelone/xCalendar/assets/78598174/1913d952-f01b-45ce-bacb-e73162ca5d33)

**Comment by Charlotte0999** [comment]
Beim UseCaseDiagram ist es etwas verwirrend mit den ganzen Pfeilen. Es wäre unserer Meinung nach etwas übersichtlicher, wenn ihr die nicht zusammenfassen könntet (rechts). Außerdem wird nicht ganz klar, wie die beiden Akteure zu unterscheiden sind. Sind das beides Anwender oder ein Anwender und ein System?
Ansonsten passt alles für uns. Supi, weiter so!
